src/test/run-make/issue-11908/Makefile

```diff
@@ -13,11 +13,11 @@ all:
 	mv $(call DYLIB,foo) $(TMPDIR)/other
 	$(RUSTC) foo.rs --crate-type=dylib
 	$(RUSTC) bar.rs -L $(TMPDIR)/other 2>&1 | \
-		grep "multiple dylib candidates"
+		grep "assertion failed" && exit 1 || exit 0
```

Comment:
No, the dylib one doesn't pass -- it prints other strange errors. The issue in question was "this causes the compiler to issue assertion failures". We could extend it to "these should work", but then we'd need to fix the other issue with dylibs.